esp-hal-common/src/usb_serial_jtag.rs

```diff
@@ -335,35 +382,43 @@ mod asynch {
     }
 
     impl UsbSerialJtag<'_> {
-        async fn write(&mut self, words: &[u8]) -> Result<(), Error> {
+        async fn write_bytes_async(&mut self, words: &[u8]) -> Result<usize, Error> {
             let reg_block = self.usb_serial.register_block();
-            for chunk in words.chunks(64 as usize) {
-                unsafe {
-                    for &b in chunk {
-                        reg_block.ep1.write(|w| w.rdwr_byte().bits(b.into()))
-                    }
-                    reg_block.ep1_conf.write(|w| w.wr_done().set_bit());
-                    UsbSerialJtagFuture::new(self.inner()).await;
+
+            let mut count = 0;
+            for chunk in words.chunks(64) {
+                for byte in chunk {
+                    reg_block
+                        .ep1
+                        .write(|w| unsafe { w.rdwr_byte().bits(*byte) });
+                    count += 1;
                 }
+                reg_block.ep1_conf.write(|w| w.wr_done().set_bit());
+
+                UsbSerialJtagFuture::new(self.inner()).await;
             }
-            Ok(())
+
+            Ok(count)
```

Comment:
Same here, I think the function will always write all the bytes (or error) so we don't need to count.